catalyst_voices/lib/widgets/buttons/voices_segmented_button.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:flutter/material.dart';
+
+/// Voices SegmentedButton implementation which defaults to flutter M3
+/// [SegmentedButton].
+///
+/// See *catalyst.dart* _buildThemeData for style configuration.
+///
+/// Be aware [as of flutter 3.22.3]:
+///   - when mixing configs of [segments], meaning only some have icons.
+///   [SegmentedButton] may incorrectly calculate segment width will cause
+///   label to split to multiline. Try avoid when possible.
+///
+/// Example:
+///
+/// ```dart
+///         VoicesSegmentedButton<int>(
+///           segments: [
+///             ButtonSegment(value: 0,label: Text('0')),
+///             ButtonSegment(value: 1,label: Text('1')),
+///             ButtonSegment(value: 2,label: Text('2')),
+///             ButtonSegment(value: 3,label: Text('3'))
+///           ],
+///           selected: _selected,
+///           onChanged: (selected) {
+///             setState(() {
+///               _selected = Set.from(selected);
+///             });
+///           },
+///           showSelectedIcon: false,
+///           emptySelectionAllowed: true,
+///         ),
+/// ```
+class VoicesSegmentedButton<T extends Object> extends StatelessWidget {
+  /// List of segments that will be shown in button
```

Comment:
Trivial but some documentation sentences end with dot, some not. I think we should add it to every sentence.